core/services/bridget/bridget.py

```diff
@@ -8,44 +9,50 @@
 from pydantic import BaseModel, conint
 from serial.tools.list_ports_linux import SysFS
 
-from settings import BridgeSettingsSpecV1, SettingsV1
+from settings import BridgeSettingsSpecV2, SettingsV2
 
+USERDATA = Path("/usr/blueos/userdata/")
 
-class BridgeSpec(BaseModel):
+
+class BridgeFrontendSpec(BaseModel):
     """Basic interface for 'bridges' links."""
 
     serial_path: str
     baud: Baudrate
     ip: str
-    udp_port: conint(gt=1023, lt=65536)  # type: ignore
+    udp_target_port: conint(ge=0, lt=65536)  # type: ignore
+    udp_listen_port: conint(ge=0, lt=65536)  # type: ignore
 
     def __str__(self) -> str:
-        return f"{self.serial_path}:{self.baud}//{self.ip}:{self.udp_port}"
+        if self.ip == "0.0.0.0":
+            return f"{self.serial_path}:{self.baud}//{self.ip}:{self.udp_listen_port}"
+        return f"{self.serial_path}:{self.baud}//{self.ip}:{self.udp_target_port}:{self.udp_listen_port}"
 
     def __hash__(self) -> int:
         return hash(str(self))
 
     @staticmethod
-    def from_settings_spec(settings_spec: BridgeSettingsSpecV1) -> "BridgeSpec":
-        return BridgeSpec(
+    def from_settings_spec(settings_spec: BridgeSettingsSpecV2) -> "BridgeFrontendSpec":
+        return BridgeFrontendSpec(
             serial_path=settings_spec.serial_path,
             baud=settings_spec.baudrate,
             ip=settings_spec.ip,
-            udp_port=settings_spec.udp_port,
+            udp_target_port=settings_spec.udp_target_port,
+            udp_listen_port=settings_spec.udp_listen_port,
         )
 
 
 class Bridget:
     """Manager for 'bridges' links."""
 
     def __init__(self) -> None:
-        self._bridges: Dict[BridgeSpec, Bridge] = {}
-        self._settings_manager = Manager("bridget", SettingsV1)
+        self._bridges: Dict[BridgeFrontendSpec, Bridge] = {}
+        self._settings_manager = Manager("bridget", SettingsV2, USERDATA / "settings" / "bridget")
```

Comment:
Why are we moving this to userdata over our standard settings folder  ?

Reply:
non-root users have no access to our regular folder because it is under /root/

Reply:
It would be nice to have a comment explaining that in the code